pwiz_tools/BiblioSpec/src/BuildParser.cpp

```diff
@@ -102,10 +102,10 @@ void BuildParser::setSpecFileName(
             }
             if (path.empty())
                 path = ".";
-            for (const auto& dir : bfs::directory_iterator(path)) {
-                bfs::path dirPath = dir.path();
-                string trialName = dirPath.filename().string();
-                for (const string& ext : extensions) {
+            for (const string& ext : extensions) { // Search for extensions in priority order

```

Comment:
I am not sure (but I might be wrong) that it is a good idea to swap the nesting of the loops iterating over extensions and folders.

For most of the extensions, we really have no opinion which one is better. People might get confused as to why the .mz5 from the grandparent directory is being used instead of the .mzML file that is in the same folder as the pepxml.

If do need to make ".mgf" that much preferred over the raw files, it might be the right thing to do would be to prevent the raw file extensions not get added to the list if it's msfragger.

Reply:
I agree with this. It's better to search all extensions in a single directory before moving to the parent directory, in the general case. And even raw file extensions should only be excluded for msfragger on a Bruker TDF.

Reply:
Well, we already had code that appeared to depend on the order of the extensions list to prioritize one flavor of mgf over another, it got tripped up because the arbitrary order of files happened to present the .d before either of the .mgf. So I swapped the loop order within the directory. The directory structure traversal should be unchanged - we still exhaust "," before moving on to "..".  Or that's what I intended anyway.

Reply:
Ah, you're right Brian. You only changed the enumeration of entries in the current directory, not the enumeration of directories. The old way was definitely dependent on the arbitrary order of entries, as you said.